tests/lib/test_openstack_security_groups.py

```diff
@@ -115,42 +112,48 @@ def test_create_security_group_forwards_result(self):
         )
         assert result == self.network_api.create_security_group.return_value
 
-    @raises(ItemNotFoundError)
     def test_create_rule_group_not_found_raises(self):
         """
         Tests that create security group rule raises if not security group is found
         """
         self.instance.find_security_group = Mock(return_value=None)
-        self.instance.create_security_group_rule(NonCallableMock(), NonCallableMock())
+        with self.assertRaises(ItemNotFoundError):
+            self.instance.create_security_group_rule(
+                NonCallableMock(), NonCallableMock()
+            )
 
-    @parameterized.expand([(1, ""), ("", 1)])
-    def test_create_rule_throws_for_missing_port(self, start_port, end_port):
+    def test_create_rule_throws_for_missing_port(self):
         """
         Tests that missing port starting values throw
         """
         mocked_details = NonCallableMock()
-        mocked_details.port_range = (start_port, end_port)
-        with assert_raises(ValueError):
-            self.instance.create_security_group_rule(NonCallableMock(), mocked_details)
+        for start, end in (0, None), (None, 0), (None, None):
+            mocked_details.port_range = (start, end)
+            with self.assertRaises(ValueError):
+                self.instance.create_security_group_rule(
+                    NonCallableMock(), mocked_details
+                )
 
-    @raises(ValueError)
     def test_create_rule_throws_for_missing_port_end(self):
         """
         Tests that missing port final values throw
         """
         mocked_details = NonCallableMock()
         mocked_details.port_range = (1, None)
-        self.instance.create_security_group_rule(NonCallableMock(), mocked_details)
+        with self.assertRaises(ValueError):
+            self.instance.create_security_group_rule(NonCallableMock(), mocked_details)
 
-    @parameterized.expand([(1, "a"), ("d", 2), ("*a", "a"), ("*", "*1")])
-    def test_create_rule_throws_non_numeric(self, start_port, end_port):
+    def test_create_rule_throws_non_numeric(self):
         """
         Tests that non-numeric inputs throw
         """
         mocked_details = NonCallableMock()
-        mocked_details.port_range = (start_port, end_port)
-        with assert_raises(ValueError):
-            self.instance.create_security_group_rule(NonCallableMock(), mocked_details)
+        for start, end in (1, "a"), ("d", 2), ("*a", "a"), ("*", "*1"):
```

Comment:
why not use `pytest.mark.parametrize`? 
so we can see which one fails if the test fails?